rules/aip0121/resource_must_support_get.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0121
+
+import (
+	"fmt"
+
+	"bitbucket.org/creachadair/stringset"
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/locations"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+// TODO: test against LROs
+var resourceMustSupportGet = &lint.ServiceRule{
+	Name: lint.NewRuleName(121, "resource-must-support-get"),
+	LintService: func(s *desc.ServiceDescriptor) []lint.Problem {
+		problems := []lint.Problem{}
+		// Add the empty string to avoid adding multiple nil checks.
+		// Just say it's valid instead.
+		resourcesWithGet := stringset.New("")
+		resourcesWithOtherMethods := map[string]*desc.MessageDescriptor{}
+		// Iterate all RPCs and try to find resources. Mark the
+		// resources which have a Get method, and which ones do not.
+		for _, m := range s.GetMethods() {
+			if utils.IsGetMethod(m) {
+				t := utils.GetResource(m.GetOutputType()).GetType()
+				resourcesWithGet.Add(t)
+			} else if utils.IsCreateMethod(m) || utils.IsUpdateMethod(m) {
+				message := utils.GetOutputOrLROResponseMessage(m)
+				if message != nil {
+					t := utils.GetResource(message).GetType()
+					resourcesWithOtherMethods[t] = message
+				}
+			} else if utils.IsListMethod(m) {
+				message := utils.GetListResourceMessage(m)
+				if message != nil {
+					t := utils.GetResource(message).GetType()
+					resourcesWithOtherMethods[t] = message
+				}
+			}
+		}
+
+		for t, m := range resourcesWithOtherMethods {
+			if !resourcesWithGet.Contains(t) {
+				problems = append(problems, lint.Problem{
+					Message: fmt.Sprintf(
+						"resource %q must support Get method", t,
+					),
+					Descriptor: m,
+					Location:   locations.MessageResource(m),
```

Comment:
If the resource is not in the diff, but the service is, this finding will not show up.

I think we want these findings to appear on the `service` itself since it is a `ServiceRule` and the service is where it will be fixed.

Reply:
ack! makes sense. I was trying to point someone in the right direction, but you're right - I forgot about tricorder's behavior not matching here.

Reply:
Actually would that work? The service will likely be unchanged as well,
so marking that line won't show up either.

Maybe on the offending method would work? presumably the offending method would
have been added to have triggered the error.

Reply:
From a general use PoV (not in Critique/a CL), where would you want to see a finding that says "Missing Standard Get RPC for resource Foo"? On the `service` that implements other RPCs related to that resource? On the resource message itself?

Then, from a Critique perspective, _when_ would you want to check for this? When a proto `service` for a resource is being authored/edited? When a resource `message` is being edited?

I think in all cases, I would personally prefer to see such a finding on/relating to the proto `service` that "looks like" it manages that resource. This is the most actionable place to emit such a finding. Emitting it in relation to the resource `message`, for example, would likely be received with confusion and "i can't fix that in this CL".

Reply:
yes, sounds good, I'll fix that first. I'm just worried about critique *not* showing the error because the service name wasn't among the modified lines of the proto.